app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -508,6 +577,24 @@ function removePendingAcceptance(minionId) {
   sessionStorage.removeItem(minionId);
 }
 
+function hasPendingRejection(minionId) {
+  return sessionStorage.getItem('reject-'+minionId) === 'true';
+}
+
+function setPendingRejection(minionId) {
+  sessionStorage.setItem('reject-'+minionId, true);
+  $('.pending-nodes-container a[data-minion-id="' + minionId + '"]').parent().text('Rejection in progress');
+}
+
+function removePendingRejection(minionId) {
+  sessionStorage.removeItem('reject-'+minionId);
+}
+
+function setPendingRemoval(minionId) {
```

Comment:
We can enhance what this method is doing like

```js
function setPendingRemoval(minionId) {
  $('.nodes-container a[data-minion-id="' + minionId + '"]').parent().text('Pending removal');
  $('#role-column-' + minionId + ' .btn').prop('disabled', true);
  $('#set-roles').data('previous', $('#set-roles').prop('disabled'));
  $('#set-roles').prop('disabled', true);
}
```

And add a method to revert the `Next` button in case the request fails. Call this in the `fail` callback. 

```js
function removePendingRemoval(minionId) {
  $('#set-roles').data('previous', $('#set-roles').prop('disabled'));
}
```